pilot/pkg/proxy/resolve.go

```diff
@@ -18,30 +18,42 @@ import (
 	"context"
 	"fmt"
 	"net"
-	"strings"
 	"time"
 
 	"istio.io/istio/pkg/log"
 )
 
-// ResolveAddr resolves an authority address to an IP address
+// ResolveAddr resolves an authority address to an IP address. Incoming
+// addr can be an IP or name. If IPv6 address, the IP part must be enclosed
+// in square brackets. Note: Lookup may return multiple IP addresses, of
```

Comment:
nit: break `note:` into its own paragraph.

Reply:
Actually, just make it a `TODO`, since it's really about the behavior of `net.DefaultResolver.LookupIPAddr` and not this method itself.

Reply:
Updating.